### PR TITLE
Fix #1544 - global codec not registered

### LINKED ISSUE #1544: Global config codec is not registered in codec cache
﻿### Expected behavior

When deserializing nested data structures the nested structure should use the codec instance assigned to the redisson config using `setCodec`.

### Actual behavior

The nested data structure is serialized using the default `JsonJacksonCodec` codec ignoring the one set on `Config` using `setCodec`.

### Steps to reproduce or test case

Create a new `JsonJacksonCodec` using a non-default `ObjectMapper` instance. 

Assign that instance to Config using `Config.setCodec`. 

Stored a nested data structure such as a `RBucketReactive` inside of a `RSetReactive`, and store a POJO inside of the `RBucketReactive` that's dependent on the configuration of the `ObjectMapper` used previously.

Attempt to retrieve the value from the `RBucketReactive` instance by iterating the `RSetReactive` contents and observe that Jackson deserialization will fail due to using a defaultly configured `ObjectMapper`.

### Redis version

4.0.10

### Redisson version

3.7.1

### Redisson configuration

All defaults except for registering a new Codec.

### PR DESCRIPTION
**Pull request by jackygurui**

### CODE REVIEW

File: redisson/src/main/java/org/redisson/Redisson.java [hunk 1]
```diff
@@ -664,8 +664,10 @@ public boolean isShuttingDown() {
         return connectionManager.isShuttingDown();
     }
 
-    protected void enableRedissonReferenceSupport() {
+    protected void enableRedissonReferenceSupport(Config config) {
         this.connectionManager.getCommandExecutor().enableRedissonReferenceSupport(this);
+        Codec codec = config.getCodec();
```
Comment (mrniko) [inline redisson/src/main/java/org/redisson/Redisson.java]: @jackygurui Please encapsulate this logic into `CommandAsyncService.enableRedissonReferenceSupport` method to have all stuff in single place.

File: redisson/src/main/java/org/redisson/RedissonReactive.java [hunk 2]
```diff
@@ -389,8 +389,10 @@ public boolean isShuttingDown() {
         return connectionManager.isShuttingDown();
     }
 
-    protected void enableRedissonReferenceSupport() {
+    protected void enableRedissonReferenceSupport(Config config) {
         this.commandExecutor.enableRedissonReferenceSupport(this);
+        Codec codec = config.getCodec();
```
Comment (mrniko) [inline redisson/src/main/java/org/redisson/RedissonReactive.java]: @jackygurui Please encapsulate this logic into `CommandAsyncService.enableRedissonReferenceSupport` method to have all stuff in single place.

File: redisson/src/main/java/org/redisson/Redisson.java (hunk 1 above)
Comment (jackygurui) [reply redisson/src/main/java/org/redisson/Redisson.java]: Good point. It is done.

File: redisson/src/main/java/org/redisson/RedissonReactive.java (hunk 2 above)
Comment (jackygurui) [reply redisson/src/main/java/org/redisson/RedissonReactive.java]: Good point. It is done.

**Comment by mrniko** [comment]
Thank you!

**Pull request merged**